src/acq.c

```diff
@@ -118,6 +120,16 @@ enum bl_error bl_acq_start(
 		acq_chan_mask = src_mask;
 	}
 
+	/* Detection mode casting to SPI mode:
+	 * reflective   -> none SPI
+	 * transmissive -> SPI mother mode
+	 * As USB is connected, SPI daughter mode is not valid
+	 */
+	if (bl_spi_mode != (enum bl_acq_spi_mode)detection_mode) {
+		bl_spi_mode = detection_mode;
+		bl_spi_init();
```

Comment:
Should we only call `bl_spi_init()` if mode is not `BL_ACQ_SPI_NONE`?

Reply:
After power up, the board is default to BL_ACQ_SPI_DAUGHTER, so it will need spi init anyway. And there is no way for the daughter board to tell if it wants to switch to other mode.

Reply:
Sorry, I misunderstood what you mean.

I think you are correct, this particular bl_spi_init should be called if in spi mode.

Reply:
It's already called in `bl_init()`, where it sets up as `BL_ACQ_SPI_DAUGHTER`, so maybe `bl_spi_init()` doesn't handle being called twice.